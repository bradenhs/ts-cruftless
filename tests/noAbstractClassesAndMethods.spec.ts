import { createSpec } from "./helpers/createSpec";

createSpec("noAbstractClassesAndMethods");
