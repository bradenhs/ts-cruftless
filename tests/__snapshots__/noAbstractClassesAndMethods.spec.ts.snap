// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`no-abstract-classes-and-methods should emit no errors when source is valid 1`] = `""`;

exports[`no-abstract-classes-and-methods should error when abstract class is present 1`] = `
"/tests/cases/noAbstractClassesAndMethods/shouldErrorWhenAbstractClassIsPresent.ts
The \\"abstract\\" keyword is not allowed no-abstract-classes-and-methods
  1 | const foo = \\"bar\\";
  2 | 
> 3 | export abstract class AbstractClass {}
    |       ^
  4 |"
`;

exports[`no-abstract-classes-and-methods should give multiple errors if abstract is found multiple times 1`] = `
"/tests/cases/noAbstractClassesAndMethods/shouldGiveMultipleErrorsIfAbstractIsFoundMultipleTimes.ts
The \\"abstract\\" keyword is not allowed no-abstract-classes-and-methods
> 1 | export abstract class Bar {
    |       ^
  2 |   abstract foo(): string;
  3 | }
  4 | 
The \\"abstract\\" keyword is not allowed no-abstract-classes-and-methods
  1 | export abstract class Bar {
> 2 |   abstract foo(): string;
    |  ^
  3 | }
  4 |"
`;
